src/server_manager/web_app/ui_components/outline-per-key-data-limit-dialog.ts

```diff
@@ -0,0 +1,341 @@
+/*
+  Copyright 2020 The Outline Authors
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import '@polymer/paper-button/paper-button';
+import '@polymer/paper-checkbox/paper-checkbox';
+import '@polymer/paper-dialog/paper-dialog';
+import '@polymer/paper-dropdown-menu/paper-dropdown-menu';
+import '@polymer/paper-input/paper-input';
+import '@polymer/paper-item/paper-item';
+import '@polymer/paper-listbox/paper-listbox';
+
+import {PaperButtonElement} from '@polymer/paper-button/paper-button';
+import {PaperDialogElement} from '@polymer/paper-dialog/paper-dialog';
+import {PaperDropdownMenuElement} from '@polymer/paper-dropdown-menu/paper-dropdown-menu';
+import {PaperInputElement} from '@polymer/paper-input/paper-input';
+import {PaperListboxElement} from '@polymer/paper-listbox/paper-listbox';
+import {css, customElement, html, internalProperty, LitElement, property} from 'lit-element';
+
+import {COMMON_STYLES} from './cloud-install-styles';
+import {DisplayAccessKey, DisplayDataAmount} from './outline-server-view';
+
+/**
+ * A floating window representing settings specific to individual access keys. Its state is
+ * dynamically set when opened using the open() method instead of with any in-HTML attributes.
+ *
+ * This element relies on conceptual separation of the data limit as input by the user, the data
+ * limit of the UI key, and the actual data limit as saved on the server.  App controls the UI data
+ * limit and the request to the server, and the display key in the element is never itself changed.
+ */
+@customElement('outline-per-key-data-limit-dialog')
+export class OutlinePerKeyDataLimitDialog extends LitElement {
+  /** @member key The UI access key representing the key we're working on. */
+  @internalProperty() key: DisplayAccessKey = null;
```

Comment:
No need to depend on DisplayAccessKey here.
Provide name and a datalimit properties instead.
The caller can do the name calculation.

Reply:
Done in c214d2a67bdc82a0c76d1542c9d6b49c07a580b0